controllers/users/checkAllowAnonDM.js

```diff
@@ -0,0 +1,21 @@
+const {User} = require('../../databases/models');
+
+module.exports = async (req, res) => {
+  let {userId} = req.params;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `userId` parameter is directly used in a database query without any validation or sanitization. This could potentially lead to security vulnerabilities such as SQL Injection attacks. Always validate and sanitize user inputs before using them in database queries.
```diff
-  let {userId} = req.params;
+  let {userId} = req.params;
+  if (!Number.isInteger(parseInt(userId))) {
+    return res.status(400).json({
+      status: 'error',
+      message: 'Invalid User ID'
+    });
+  }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->